src/main/java/org/folio/consortia/service/impl/SharingInstanceServiceImpl.java

```diff
@@ -44,13 +57,50 @@ public SharingInstance getById(UUID consortiumId, UUID actionId) {
   @Override
   @Transactional
   public SharingInstance start(UUID consortiumId, SharingInstance sharingInstance) {
-    log.debug("start:: Trying to start instance sharing: {} for consortium: {}", sharingInstance.getInstanceIdentifier(), consortiumId);
+    log.debug("start:: Trying to start instance sharing with instanceIdentifier: {}, consortiumId: {}.", sharingInstance.getInstanceIdentifier(), consortiumId);
     consortiumService.checkConsortiumExistsOrThrow(consortiumId);
-    checkTenantsExistAndContainCentralTenantOrThrow(sharingInstance.getSourceTenantId(), sharingInstance.getTargetTenantId());
 
-    SharingInstanceEntity savedSharingInstance = sharingInstanceRepository.save(toEntity(sharingInstance));
-    log.info("start:: sharingInstance with id: {}, instanceId: {}, sourceTenantId: {}, targetTenantId: {} successfully saved in db",
-      savedSharingInstance.getId(), savedSharingInstance.getInstanceId(), savedSharingInstance.getSourceTenantId(), savedSharingInstance.getTargetTenantId());
+    String centralTenantId = tenantService.getCentralTenantId();
+    String sourceTenantId = sharingInstance.getSourceTenantId();
+    String targetTenantId = sharingInstance.getTargetTenantId();
+    checkTenantsExistAndContainCentralTenantOrThrow(sourceTenantId, targetTenantId);
+
+    SharingInstanceEntity savedSharingInstance;
+    if (Objects.equals(centralTenantId, sourceTenantId)) {
+      FolioExecutionContext currentTenantContext = (FolioExecutionContext) folioExecutionContext.getInstance();
+      JsonNode inventoryInstance;
+
+      try (var context = new FolioExecutionContextSetter(prepareContextForTenant(sourceTenantId, folioModuleMetadata, currentTenantContext))) {
+        inventoryInstance = inventoryService.getById(sharingInstance.getInstanceIdentifier());
+      } catch (Exception ex) {
+        sharingInstance.setStatus(Status.ERROR);
+        sharingInstance.setError(String.format("Failed to get inventory instance with reason: %s", InventoryClient.getReason(ex)));
+        savedSharingInstance = sharingInstanceRepository.save(toEntity(sharingInstance));
+        log.info("start:: sharingInstance with instanceId: {}, sourceTenantId: {}, targetTenantId: {} failed with reason: {}",
+          sharingInstance.getInstanceIdentifier(), sourceTenantId, targetTenantId, sharingInstance.getError());
+        return converter.convert(savedSharingInstance, SharingInstance.class);
+      }
+
+      try (var context = new FolioExecutionContextSetter(prepareContextForTenant(targetTenantId, folioModuleMetadata, currentTenantContext))) {
+        String source = "folio".equalsIgnoreCase(inventoryInstance.get("source").asText()) ? "CONSORTIUM-FOLIO" : "CONSORTIUM-MARC";
```

Comment:
is it expected to have these hardcoded values here?

Reply:
1. We have only two possible cases for shadow instance “CONSORTIUM-FOLIO” or “CONSORTIUM-MARC”. So currently I am setting former one, but yes it is not final yet.
2. _source_ field is mandatory in _'inventory/instance'_ https://s3.amazonaws.com/foliodocs/api/mod-inventory/r/inventory.html#inventory_instances_post:~:text=%22required%22%3A%20%5B%0A%20%20%20%20%22source%22%2C